pkg/resolver/search.go

```diff
@@ -196,6 +198,88 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
 	return items, nil
 }
 
+// Remove operator (<=, >=, !=, !, <, >, =) if any from values
+func getOperator(values []string) (string, []string) {
+	// Get the operator (/^<=|^>=|^!=|^!|^<|^>|^=/)
+	var operator string
+	// Replace any of these symbols with ""
+	replacer := strings.NewReplacer("<=", "",
+		">=", "",
+		"!=", "",
+		"!", "",
+		"<", "",
+		">", "",
+		"=", "")
+	newValues := make([]string, len(values)) // To store cleaned up values (without operator)
+	for i, value := range values {
+		operatorRemovedValue := replacer.Replace(value)
+		newValues[i] = operatorRemovedValue
+		operator = strings.Replace(value, operatorRemovedValue, "", 1) // find operator
+	}
+	return operator, newValues
+}
+
+func getWhereClauseExpression(prop, operator string, values []string) exp.Expression {
+	switch operator {
+	case "<=":
+		return goqu.L(`"data"->>?`, prop).Lte(values).Expression()
+	case ">=":
+		return goqu.L(`"data"->>?`, prop).Gte(values).Expression()
+	case "!=":
+		return goqu.L(`"data"->>?`, prop).Neq(values).Expression()
+	case "!":
+		return goqu.L(`"data"->>?`, prop).NotIn(values).Expression()
+	case "<":
+		return goqu.L(`"data"->>?`, prop).Lt(values).Expression()
+	case ">":
+		return goqu.L(`"data"->>?`, prop).Gt(values).Expression()
+	case "=":
+		return goqu.L(`"data"->>?`, prop).In(values).Expression()
+	default:
+		if prop == "cluster" {
+			return goqu.C(prop).In(values).Expression()
+		}
+		return goqu.L(`"data"->>?`, prop).In(values).Expression()
+	}
+
+}
+
+func getDateFilter(values []string) (string, []string) {
+	// Expected values: {"hour", "day", "week", "month", "year"}
+	newValues := make([]string, len(values))
+
+	now := time.Now()
+	operator := ">"
+	for i, val := range values {
+		switch val {
+		case "hour":
+			then := now.Add(time.Duration(-1) * time.Hour).Format("2006-01-02T15:04:05Z")
+			newValues[i] = then
+
+		case "day":
+			then := now.AddDate(0, 0, -1).Format("2006-01-02T15:04:05Z")
+			newValues[i] = then
```

Comment:
We could reduce these to a single line.
```suggestion
			newValues[i] = now.AddDate(0, 0, -1).Format("2006-01-02T15:04:05Z")
```

Reply:
Updated